### DISCUSSION TITLE
Being able to set the display column per table

### DISCUSSION THREAD

**Discussion by thinkbig1979** [teableio/teable#732] [Ideas]
After about an hour of playing with Teable, I have to say, it has great potential. A lot of things have been tackled really nicely.

Once thing I ran into was that currently, the primary key column is automatically used as the display value in linked columns in other tables. 

-----
Example: 
I have a table of contacts and the `First Name` column is the primary key in that table. I also have a `Surname` column and a formula column called `Full Name` in that table that shows the concatenated first name and surname. 

I also have a `Companies` table, with an `Employees` column, which links to the `Contacts` table. 
The issue is that only the first names of the contacts show up in the `Employees` column, because `First Name` is the primary key column for the `Contacts` table.

----

I know it's possible to create a lookup column to display the full names, but them you lose the ability to click on the names in the column and get the popup with the entries from the other table. It would be great to be able to see the `Full Name` field reflected in the `Employees` column, while still maintaining the ability to click on the name and get the popup with the linked table.

Budibase, for example, allows you to set any column in a table as the default display column, including formula columns. The downside to Budibase's implementation is that you cannot search formula fields properly, so if you then try to search for an employee name in the Companies table, it does not work properly. I know companies have moved away from Budibase over that issue. 

If Teable could solve both these issues:
- Set which column in a table is shown in linked tables
- Ensure that full text search works properly in formula columns

...that would be awesome.

**Comment by tea-artist** [comment]
Teable allows the primary column to be set as a formula, and you can directly convert the primary column into a formula for fullName, and the formula column naturally supports search.
Hope it helps.

**Comment by thinkbig1979** [comment]
Lesson: When you create a new table, one column is always marked as the primary key (it has the lock symbol alongside the column name). Regardless of how you arrange the columns, that column will always be used as the display column in linked tables, so if you want to see something specific in other tables, this is the column you need to set up the way you want. It is also possible for it to be a formula column, that is also fully searchable in other tables. 
COOL!
